### ISSUE TITLE
Retry on http502 error during OWSlib request

### ISSUE THREAD

**Issue by jochemthart1** [geopython/GeoHealthCheck#386]
_This issue is related to issue #366._

**Is your feature request related to a problem? Please describe.**
During a stress test of GHC (lots of resources with high frequency), I am getting mysterious http502 errors during `WebFeatureService` request in the `get_metadata` function inside `wfs.py`, `wms.py`, etc. These errors can be avoided by retrying requests.

**Describe the solution you'd like**
Currently, in [probe.py](https://github.com/geopython/GeoHealthCheck/blob/master/GeoHealthCheck/probe.py#L94) there is already something in place for this using the `create_requests_retry_session()` function in [util.py](https://github.com/geopython/GeoHealthCheck/blob/master/GeoHealthCheck/util.py#L267). The same principle could be applied to the request that is done in `get_metadata` functions of each probe (wfs.py, wms.py, etc.). 
[wfs.py](https://github.com/geopython/GeoHealthCheck/blob/master/GeoHealthCheck/plugins/probe/wfs.py#L123)

**Describe alternatives you've considered**
An alternative is a simple loop with an exception clause which retries the `WebFeatureService` request up to 3 times:
```
    def get_metadata(self, resource, version='1.1.0', retries=3):
        """
        Get metadata, specific per Resource type.
        :param resource:
        :param version:
        :return: Metadata object
        request_headers = self.get_request_headers()
        """
        for i in range(retries):
            try:
                return WebFeatureService(
                    resource.url,
                    version=version,
                    headers=request_headers)
            except:
                continue
        
        raise Exception(f'Get Metadata Request Exceeded {retries} Retries')
```

**Comment by justb4** [comment]
Yes, best to fix with a "retry session" in OWSLib. Though, there can always be cases where there is really a problem: for example a load-balancer with e.g. 3 backend GeoServer instances of which 2 are permanently failing for some reason. Then 3 retries will always succeed, but there is still an undetected problem. Maybe a "Warning" or "Suspicious" type of verdict would be better in those cases.  Or are these 502-cases caused within the GHC-Docker environment somehow?
